### PR TITLE
[doc][1LP][RFR][NOTEST] Some coding style suggestions.

### PR DESCRIPTION
**Pull request by quarckster**
Intent
=================

This is a suggestion for our coding style.

### CODE REVIEW

File: docs/guides/dev_guide.rst [hunk 1]
```diff
@@ -184,6 +204,13 @@ We also do a few things that aren't explicitly called out in PEP 8:
 
     "{0} and another {0}".format("something")
 
+* There is a one exception for string formatting. According `<https://docs.python.org/3/howto/logging.html#optimization>`_
+  use old style ``%s``::
```
Comment (mfalesni) [inline docs/guides/dev_guide.rst]: "..., but without the actual `%` formatting operation"

File: docs/guides/dev_guide.rst [hunk 2]
```diff
@@ -84,20 +84,40 @@ We also do a few things that aren't explicitly called out in PEP 8:
...
 
+* Use parentheses `()` for line continuation::
+
+    # in imports
+    import (module1, module2, module3, module4,
+        module5)
+
+    # for long strings
+    some_long_sting = ("Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod "
+        "tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam")
```
Comment (mfalesni) [inline docs/guides/dev_guide.rst]: I personally hate this kind of separation, the one below is much more readable.

File: docs/guides/dev_guide.rst (hunk 1 above)
Comment (quarckster) [inline docs/guides/dev_guide.rst]: Added

File: docs/guides/dev_guide.rst (hunk 2 above)
Comment (quarckster) [inline docs/guides/dev_guide.rst]: Removed

File: docs/guides/dev_guide.rst [hunk 3]
```diff
@@ -197,6 +222,7 @@ General Notes
 * If you feel icky about something you've written but don't know how to make
   it better, ask someone. It's better to have it fixed before submitting it as
   a pull request ;)
+* Use :py:func:`six` library to write Python 3 compatible code.
```
Comment (mfalesni) [inline docs/guides/dev_guide.rst]: `:py:mod:` #nitpick

Comment (quarckster) [inline docs/guides/dev_guide.rst]: Done

File: docs/guides/dev_guide.rst [hunk 4]
```diff
@@ -84,20 +84,37 @@ We also do a few things that aren't explicitly called out in PEP 8:
   width of 100 characters. As a result, our maximum line length is 100 characters,
   rather than 80.
 
+* Use parentheses ``()`` for line continuation::
+
+    # in imports
+    import (module1, module2, module3, module4,
```
Comment (mshriver) [inline docs/guides/dev_guide.rst]: I think most of our current imports that use () do so with this style:

```python
import (
    module1, module2, module3,
    module4 )

Comment (quarckster) [inline docs/guides/dev_guide.rst]: This looks a bit ugly to me, but most important that parentheses are used here.

File: docs/guides/dev_guide.rst [hunk 5]
```diff
@@ -84,20 +84,37 @@ We also do a few things that aren't explicitly called out in PEP 8:
...
+        module5)
+
+    # for long strings
```
Comment (izapolsk) [inline docs/guides/dev_guide.rst]: I'd use doc strings for such case.

Comment (quarckster) [inline docs/guides/dev_guide.rst]: Docstrings can be used when string contains multiple lines, but this case for long string without multiple lines.

Comment (mshriver) [inline docs/guides/dev_guide.rst]: $.02: Agree with docstrings for multi-line, `()` for single line, discourage use of '\n' in strings.

**Pull request merged**